langchain/src/agents/executor.ts

```diff
@@ -72,11 +72,13 @@ export class AgentExecutor extends BaseChain {
 
     const getOutput = async (finishStep: AgentFinish) => {
       const { returnValues } = finishStep;
+      const additional = await this.agent.prepareForOutput(returnValues, steps);
+
       if (this.returnIntermediateSteps) {
-        return { ...returnValues, intermediateSteps: steps };
+        return { ...returnValues, intermediateSteps: steps, ...additional };
       }
       await this.callbackManager.handleAgentEnd(finishStep, this.verbose);
```

Comment:
Seems likely youd want to track these in the trace as well? not sure though havent looked at this new stuff yet